src/main/java/uk/gov/pay/api/resources/AgreementsResource.java

```diff
@@ -39,7 +39,7 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(AgreementsResource.class);
 
     private final String baseUrl;
-    private AgreementService agreementService;
+    private final AgreementService agreementService;
 
```

Comment:
I know it's not your fault, but `import uk.gov.pay.api.model.links.PaymentWithAllLinks;
` is unused.